src/setups/rectangular_tank.jl

```diff
@@ -149,27 +149,33 @@ struct RectangularTank{NDIMS, NDIMSt2, ELTYPE <: Real}
                                                               particle_spacing,
                                                               n_particles_per_dim)
 
-        if state_equation !== nothing
-            # Use hydrostatic pressure gradient and calculate density from inverse state
-            # equation, so don't pass fluid density.
-            fluid = RectangularShape(particle_spacing, n_particles_per_dim, zeros(NDIMS);
-                                     velocity, pressure, acceleration, state_equation,
-                                     mass=fluid_mass)
-        else
-            fluid = RectangularShape(particle_spacing, n_particles_per_dim, zeros(NDIMS);
-                                     density=fluid_density, velocity, pressure,
-                                     acceleration, state_equation, mass=fluid_mass)
-        end
-
         boundary = InitialCondition(coordinates=boundary_coordinates,
                                     velocity=boundary_velocities,
                                     mass=boundary_masses, density=boundary_densities,
                                     particle_spacing=boundary_spacing)
 
         # Move the tank corner in the negative coordinate directions to the desired position
-        fluid.coordinates .+= min_coordinates
         boundary.coordinates .+= min_coordinates
 
+        fluid = boundary
+        if norm(fluid_size) > eps()
+            if state_equation !== nothing
+                # Use hydrostatic pressure gradient and calculate density from inverse state
+                # equation, so don't pass fluid density.
+                fluid = RectangularShape(particle_spacing, n_particles_per_dim,
+                                         zeros(NDIMS);
+                                         velocity, pressure, acceleration, state_equation,
+                                         mass=fluid_mass)
+            else
+                fluid = RectangularShape(particle_spacing, n_particles_per_dim,
+                                         zeros(NDIMS);
+                                         density=fluid_density, velocity, pressure,
+                                         acceleration, state_equation, mass=fluid_mass)
+            end
+            # Move the tank corner in the negative coordinate directions to the desired position
+            fluid.coordinates .+= min_coordinates
+        end
+
```

Comment:
What is happening here? If the fluid size is zero, return the boundary twice? That's awkward.

Reply:
I use the tank without a fluid inside.

Reply:
But that's a really awkward way to do that. Can't you just return an empty `InitialCondition` for the fluid in this case?